test/integration/spec/remotetracks.js

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable no-undefined */
+'use strict';
+
+const assert = require('assert');
+const { trackPriority } = require('../../../lib/util/constants');
+const { trackPriority: { PRIORITY_HIGH, PRIORITY_LOW, PRIORITY_STANDARD } } = require('../../../lib/util/constants');
+const LocalDataTrack = require('../../../lib/media/track/es5/localdatatrack');
+const defaults = require('../../lib/defaults');
+const { completeRoom } = require('../../lib/rest');
+const { audio: createLocalAudioTrack, video: createLocalVideoTrack } = require('../../../lib/createlocaltrack');
+const {
+  createSyntheticAudioStreamTrack,
+  setup,
+  smallVideoConstraints,
+  tracksSubscribed,
+  trackSwitchedOff,
+  trackSwitchedOn,
+  waitFor
+} = require('../../lib/util');
+
+function getTracksOfKind(participant, kind) {
+  return [...participant.tracks.values()].filter(remoteTrack => remoteTrack.kind !== kind).map(({ track }) => track);
+}
+
+describe('RemoteParticipant', function() {
+  // eslint-disable-next-line no-invalid-this
+  this.timeout(60000);
+
+  describe('.setPriority', () => {
+    let thisRoom; let thoseRooms;
+    let aliceLocal; let bobLocal;
+    let aliceRemote; let bobRemote;
+    let aliceTracks; let bobTracks;
+    let aliceRemoteVideoTrack; let bobRemoteVideoTrack;
+
+    beforeEach(async () => {
+      const dataTrack = new LocalDataTrack();
+      [, thisRoom, thoseRooms] = await setup({
+        testOptions: {
+          bandwidthProfile: {
+            video: { maxTracks: 1 }
+          },
+          tracks: [dataTrack]
+        },
+        otherOptions: { tracks: [dataTrack] },
+        nTracks: 0
+      });
+
+      [aliceTracks, bobTracks] = await Promise.all([1, 2, 3].map(async () => [
+        createSyntheticAudioStreamTrack() || await createLocalAudioTrack({ fake: true }),
+        await createLocalVideoTrack(smallVideoConstraints),
+      ]));
+
+      [aliceLocal, bobLocal] = thoseRooms.map(room => room.localParticipant);
+      [aliceRemote, bobRemote] = [thisRoom.participants.get(aliceLocal.sid), thisRoom.participants.get(bobLocal.sid)];
+
+      // Alice publishes her tracks at low priority
+      // Bob publishes his tracks at medium priority
+      await Promise.all([
+        ...aliceTracks.map(track => aliceLocal.publishTrack(track, { priority: PRIORITY_LOW })),
+        ...bobTracks.map(track => bobLocal.publishTrack(track, { priority: PRIORITY_STANDARD })),
+        tracksSubscribed(aliceRemote, 3),
+        tracksSubscribed(bobRemote, 3)
+      ]);
+
+      [aliceRemoteVideoTrack, bobRemoteVideoTrack] = [aliceRemote, bobRemote].map(({ videoTracks }) => {
+        return [...videoTracks.values()][0].track;
+      });
+    });
+
+    afterEach(async () => {
+      [thisRoom, ...thoseRooms].forEach(room => room && room.disconnect());
+      if (thisRoom) {
+        await completeRoom(thisRoom.sid);
+      }
+    });
+
+    ['audio', 'video', 'data'].forEach(kind => {
```

Comment:
@makarandp0 
```js
combinationContext([
  [
    ['audio', 'data', 'video'],
    x => `when called on a Remote${capitalize(x)}Track`
  ],
  [
    [null, ...Object.values(trackPriority), undefined, 'foo', 42, true, {}, []]
    x => `when called with priority ${x}`
  ]
], ([kind, priority]) => {
  it(`should return the Remote${capitalize(kind)}Track`, () => {
    ...
  });
  
  if (![null, ...Object.values(trackPriority)].includes(priority) {
    it(`should set .priority`, () => {
      ..
    });
  } else {
    it(`should not set .priority`, () => {
      ..
    });
  }
});
```

Reply:
@manjeshbhargav - I think separating the valid and invalid test cases makes it much more readable. There isn't much duplication of code to warrant consolidating them.